patches/minecraft/net/minecraft/client/particle/ParticleManager.java.patch

```diff
@@ -39,7 +39,17 @@
        this.field_187241_h.add(p_78873_1_);
     }
  
-@@ -342,7 +343,7 @@
+@@ -306,7 +307,8 @@
+       Particle.field_70554_ao = p_215233_1_.func_216785_c().field_72448_b;
+       Particle.field_70555_ap = p_215233_1_.func_216785_c().field_72449_c;
+ 
+-      for(IParticleRenderType iparticlerendertype : field_215241_b) {
++      for(IParticleRenderType iparticlerendertype : this.field_78876_b.keySet()) { // Forge: allow custom IParticleRenderType's
```

Comment:
Random nitpick: The apostrophe in `IParticleRenderType's` shouldn't be there as plurals do not use apostrophes.

Reply:
The apostrophy was likely added to separate the pluralization from the proper class name.

Reply:
^